nutils/evaluable.py

```diff
@@ -766,7 +766,7 @@ def __get__(self, instance, owner):
         def bound_method_wrapper(*args, **kwargs):
           res = bound_orig(*args, **kwargs)
           if hasattr(instance, 'dtype'):
-            assert res.dtype == instance.dtype, ((instance.dtype, res.dtype), instance, res)
+            assert numpy.can_cast(res.dtype, instance.dtype, casting='safe'), ((instance.dtype, res.dtype), instance, res)
```

Comment:
If `can_cast` is the way to go, we still want to know which casting rule to use, 'safe', 'same_kind', etc.

Reply:
Indeed - I'm thinking `same_kind` is the one we want

Reply:
See [here](https://github.com/soraros/nutils/pull/3#discussion_r548080402).